File: backend-pacientes/src/entities/TipoIdentificacion.ts
import { Entity, Column, PrimaryColumn, OneToMany } from "typeorm";
import { Paciente } from "./Paciente";

@Entity("DAF_TIPOS_IDENTIFICACION")
export class TipoIdentificacion {

  @PrimaryColumn({ name: "CODIGO_TIPO_IDENTIFICACION" })
  codigoTipoIdentificacion!: string;

  @Column({ name: "NOMBRE_TIPO_IDENTIFICACION" })
  nombreTipoIdentificacion!: string;

  @Column({ name: "ESTADO" })
  estado!: string;

  @OneToMany(() => Paciente, paciente => paciente.tipoIdentificacion)
  pacientes!: Paciente[];
}

File: pacientes-api/src/main/java/com/veris/pacientes_api/entity/TipoIdentificacion.java
@Entity
@Table(name = "DAF_TIPOS_IDENTIFICACION")
@Data
public class TipoIdentificacion {

    @Id
    @Column(name = "CODIGO_TIPO_IDENTIFICACION")
    private String codigoTipoIdentificacion;

    @Column(name = "NOMBRE_TIPO_IDENTIFICACION")
    private String nombreTipoIdentificacion;

    @Column(name = "ESTADO")
    private String estado;
}
